discord/ui/select.py

```diff
@@ -343,12 +344,29 @@ def is_dispatchable(self) -> bool:
         return True
 
     @classmethod
-    def from_component(cls, component: SelectMenu) -> Self:
-        return cls(
-            **{k: getattr(component, k) for k in cls.__item_repr_attributes__},
-            custom_id=component.custom_id,
-            row=None,
-        )
+    def from_component(cls, component: SelectComponent) -> Self:
+        if not isinstance(component, SelectComponent):
+            raise TypeError('Excepted an instance of SelectComponent not {component.__class__.__name__}')
+
+        kwrgs = {
+            'custom_id': component.custom_id,
+            'placeholder': component.placeholder,
+            'min_values': component.min_values,
+            'max_values': component.max_values,
+            'disabled': component.disabled,
+        }
```

Comment:
This is probably better off being a separate attribute, such as `__from_component_attributes__` with a tuple of attributes or something in each one similar to the old design but decoupled from the `__repr__`.

Reply:
Again with the `BaseSelect.... += (..., ...)` ?

Reply:
Not `+=` because that modifies the original but yes. The problem with the old one was that it was coupled with `__repr__` which made the output way too verbose.